pkg/nfd-gc/nfd-gc.go

```diff
@@ -126,26 +153,46 @@ func (n *topologyGC) garbageCollect() {
 		nodeNames.Insert(node.Name)
 	}
 
-	nrts, err := n.topoClient.TopologyV1alpha2().NodeResourceTopologies().List(context.TODO(), metav1.ListOptions{})
-	if err != nil {
-		klog.ErrorS(err, "failed to list NodeResourceTopology objects")
-		return
+	// Handle NodeFeature objects
+	nfs, err := n.nfdClient.NfdV1alpha1().NodeFeatures("").List(context.TODO(), metav1.ListOptions{})
+	if errors.IsNotFound(err) {
+		klog.V(2).InfoS("NodeFeature CRD does not exist")
+	} else if err != nil {
+		klog.ErrorS(err, "failed to list NodeFeature objects")
+	} else {
+		for _, nf := range nfs.Items {
+			nodeName, ok := nf.GetLabels()[nfdv1alpha1.NodeFeatureObjNodeNameLabel]
+			if !ok {
+				klog.InfoS("node name label missing from NodeFeature object", "nodefeature", klog.KObj(&nf))
+			}
+			if !nodeNames.Has(nodeName) {
+				n.deleteNodeFeature(nf.Namespace, nf.Name)
+			}
+		}
 	}
 
-	for _, nrt := range nrts.Items {
-		key, err := cache.MetaNamespaceKeyFunc(&nrt)
-		if err != nil {
-			klog.ErrorS(err, "failed to create key", "noderesourcetopology", klog.KObj(&nrt))
-			continue
-		}
-		if !nodeNames.Has(key) {
-			n.deleteNRT(key)
+	// Handle NodeResourceTopology objects
+	nrts, err := n.topoClient.TopologyV1alpha2().NodeResourceTopologies().List(context.TODO(), metav1.ListOptions{})
+	if errors.IsNotFound(err) {
+		klog.V(2).InfoS("NodeResourceTopology CRD does not exist")
+	} else if err != nil {
+		klog.ErrorS(err, "failed to list NodeResourceTopology objects")
+	} else {
+		for _, nrt := range nrts.Items {
+			key, err := cache.MetaNamespaceKeyFunc(&nrt)
+			if err != nil {
+				klog.ErrorS(err, "failed to create key", "noderesourcetopology", klog.KObj(&nrt))
+				continue
+			}
+			if !nodeNames.Has(key) {
+				n.deleteNRT(key)
+			}
```

Comment:
```suggestion
			if !nodeNames.Has(nrt.Name) {
				n.deleteNRT(key)
			}
```

Reply:
Thanks @PiotrProkop, makes total sense (I was blindly moving code without thinking much). I incorporated this change in the ` nfd-gc: support garbage collection of NodeFeatures` patch